google-cloud-storage/src/main/java/com/google/cloud/storage/MultipartUploadHttpRequestManager.java

```diff
@@ -188,8 +188,12 @@ CompleteMultipartUploadResponse sendCompleteMultipartUploadRequest(
     if (request.userProject() != null) {
       httpRequest.getHeaders().put("x-goog-user-project", request.userProject());
     }
-    @Nullable Crc32cLengthKnown crc32cValue = Hasher.defaultHasher().hash(ByteBuffer.wrap(bytes));
-    addChecksumHeader(crc32cValue, httpRequest.getHeaders());
+    if (request.getCrc32c() != null) {
+      addChecksumHeader(request.getCrc32c(), httpRequest.getHeaders());
+    } else {
+      @Nullable Crc32cLengthKnown crc32cValue = Hasher.defaultHasher().hash(ByteBuffer.wrap(bytes));
```

Comment:
Can this be NULL?

If that is the case you shouldn't remove the null check below. Right now its a contradiction.